onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -1116,10 +1156,98 @@ def aten_bitwise_or(self: TInt, other: TInt) -> TInt:
 
 
 @torch_op("aten::bitwise_right_shift")
-def aten_bitwise_right_shift(self: TInt, other: TInt) -> TInt:
+def aten_bitwise_right_shift_int16(self: INT16, other: INT16) -> INT16:
+    """bitwise_right_shift.Tensor(Tensor self, Tensor other) -> Tensor"""
+    negative = op.Less(self, 0)
+    self = op.Cast(self, to=UINT16.dtype)
+    other = op.Cast(other, to=UINT16.dtype)
+
+    # Simulate signed bitshift with unsigned bitshift
+    # Clear the lower bits for the mask by shifting right then left
+    mask = op.BitShift(
+        op.Cast(op.Constant(value_int=0xFFFF), to=UINT16.dtype), other, direction="LEFT"
+    )
+    mask = op.BitShift(mask, other, direction="RIGHT")
+    mask = op.BitwiseNot(mask)
+    # Do unsigned shift
+    shifted = op.BitShift(self, other, direction="RIGHT")
+    negative_shifted = op.BitwiseOr(shifted, mask)
+    # Fill in the upper bits with the mask if the sign bit was set
+    return op.Where(
+        negative, op.Cast(negative_shifted, to=INT16.dtype), op.Cast(shifted, to=INT16.dtype)
+    )
+
+
+@torch_op("aten::bitwise_right_shift")
+def aten_bitwise_right_shift_int32(self: INT32, other: INT32) -> INT32:
+    """bitwise_right_shift.Tensor(Tensor self, Tensor other) -> Tensor"""
+    negative = op.Less(self, 0)
+    self = op.Cast(self, to=UINT32.dtype)
+    other = op.Cast(other, to=UINT32.dtype)
+
+    # Simulate signed bitshift with unsigned bitshift
+    # Clear the lower bits for the mask by shifting right then left
```

Comment:
Can you clarify (preferably in comments in code) the desired behavior? I guess there are just two options, right, either preserving the sign bit or using zeros?

Reply:
Are you trying to simulate arithmetic shift using logical shift?

Reply:
> Are you trying to simulate arithmetic shift using logical shift?

Yes. What you said is more accurate

Reply:
Hopefully it's clearer now?